### PR TITLE
Feature/7842 apim smart detector alert rule

### PR DESCRIPTION
**Pull request by ArtiomMatiom**
Now it should be possible to create alert rules when creating a APIM resource

### CODE REVIEW

File: modules/api_management/main.tf [hunk 1]
```diff
@@ -136,6 +136,22 @@ resource "azurerm_api_management_diagnostic" "apim_diagnostic" {
   }
 }
 
+resource "azurerm_monitor_smart_detector_alert_rule" "example" {
+  count = length(var.alert_rules_settings)
+
```
Comment (jvanenckevort) [inline modules/api_management/main.tf:141]: I think the line breaks can be removed here

File: modules/api_management/variables.tf [hunk 2]
```diff
@@ -80,6 +80,20 @@ variable "azurerm_api_management_diagnostic_settings" {
   }
 }
 
+variable "alert_rules_settings" {
+  type = list(object({
+    name          = string,
+    severity      = string,       # possible values: Sev0, Sev1, Sev2, Sev3 or Sev4
+    frequency     = string,       # frequency in ISO-8601 eg. PT1M -> 1 per Month
+    detector_type = string,       # possible values: FailureAnomaliesDetector, RequestPerformanceDegradationDetector, DependencyPerformanceDegradationDetector, ExceptionVolumeChangedDetector, TraceSeverityDetector, MemoryLeakDetector
```
Comment (jvanenckevort) [inline modules/api_management/variables.tf:88]: This comment is very long, maybe provide a link to the possible values?

**Review by jvanenckevort — CHANGES_REQUESTED** [review]
I put in a few minor comments to resolve

File: modules/api_management/variables.tf (hunk 2 above)
Comment (ArtiomMatiom) [reply modules/api_management/variables.tf:88]: Ok, provided the link to terraform.registery

File: modules/api_management/main.tf (hunk 1 above)
Comment (ArtiomMatiom) [reply modules/api_management/main.tf:141]: Remove empty lines

File: modules/api_management/variables.tf [hunk 3]
```diff
@@ -70,7 +70,7 @@ variable "azurerm_api_management_diagnostic_settings" {
     verbosity                 = string, # possible values: verbose, information, error
     http_correlation_protocol = string, # possible values: None, Legacy, W3C
   })
-  description = "Settings for api management diagnostic, will be created only if api_management_logger_settings have been provided."
+  description = "Settings for api management diagnostic, api_management-diagnostic will be created only if api_management_logger_settings have been provided. "
```
[the remark this answers is not in the document]
Comment (ArtiomMatiom) [reply modules/api_management/variables.tf:73]: indeed switched to hyphen

File: modules/api_management/main.tf [hunk 4]
```diff
@@ -136,6 +136,20 @@ resource "azurerm_api_management_diagnostic" "apim_diagnostic" {
   }
 }
 
+resource "azurerm_monitor_smart_detector_alert_rule" "example" {
+  count = length(var.alert_rules_settings)
```
Comment (jvanenckevort) [inline modules/api_management/main.tf:140]: Missing fmt formatter ;) all the = signs are alligned if you use the formatter

**Review by jvanenckevort — CHANGES_REQUESTED** [review]

File: modules/api_management/main.tf (hunk 4 above)
Comment (ArtiomMatiom) [reply modules/api_management/main.tf]: Ah, :) my bad

**Review by jvanenckevort — APPROVED** [review]

**Pull request merged**